.gitlab-ci.yml

```diff
@@ -1,37 +1,591 @@
+    - git clone git@github.com:mendersoftware/tenantadm go/src/github.com/mendersoftware/tenantadm || true
+    - (cd go/src/github.com/mendersoftware/tenantadm && git fetch -u -f origin ${TENANTADM_REV}:pr && git checkout pr)
+    - (cd go/src/github.com/mendersoftware/tenantadm && echo TENANTADM_REV_GIT_SHA=$(git rev-parse HEAD) >> ${CI_PROJECT_DIR}/build_revisions.env)
+    - git clone https://github.com/mem/oe-meta-go.git || true
+    - (cd oe-meta-go && git checkout -f master)
+    - git clone git://git.openembedded.org/meta-openembedded.git || true
+    - (cd meta-openembedded && git fetch -u -f origin ${META_OPENEMBEDDED_REV}:pr && git checkout pr)
+    - (cd meta-openembedded && echo META_OPENEMBEDDED_REV_GIT_SHA=$(git rev-parse HEAD) >> ${CI_PROJECT_DIR}/build_revisions.env)
+    - git clone git://github.com/agherzan/meta-raspberrypi.git || true
+    - (cd meta-raspberrypi && git fetch -u -f origin ${META_RASPBERRYPI_REV}:pr && git checkout pr)
+    - (cd meta-raspberrypi && echo META_RASPBERRYPI_REV_GIT_SHA=$(git rev-parse HEAD) >> ${CI_PROJECT_DIR}/build_revisions.env)
+    - git clone git://github.com/mendersoftware/mender-conductor.git go/src/github.com/mendersoftware/mender-conductor || true
+    - (cd go/src/github.com/mendersoftware/mender-conductor && git fetch -u -f origin ${MENDER_CONDUCTOR_REV}:pr && git checkout pr)
+    - (cd go/src/github.com/mendersoftware/mender-conductor && echo MENDER_CONDUCTOR_REV_GIT_SHA=$(git rev-parse HEAD) >> ${CI_PROJECT_DIR}/build_revisions.env)
+    - git clone git@github.com:mendersoftware/mender-conductor-enterprise.git go/src/github.com/mendersoftware/mender-conductor-enterprise || true
+    - (cd go/src/github.com/mendersoftware/mender-conductor-enterprise && git fetch -u -f origin ${MENDER_CONDUCTOR_ENTERPRISE_REV}:pr && git checkout pr)
+    - (cd go/src/github.com/mendersoftware/mender-conductor-enterprise && echo MENDER_CONDUCTOR_ENTERPRISE_REV_GIT_SHA=$(git rev-parse HEAD) >> ${CI_PROJECT_DIR}/build_revisions.env)
+    - git clone https://github.com/mendersoftware/mender-cli.git go/src/github.com/mendersoftware/mender-cli || true
+    - (cd go/src/github.com/mendersoftware/mender-cli && git fetch -u -f origin ${MENDER_CLI_REV}:pr && git checkout pr)
+    - (cd go/src/github.com/mendersoftware/mender-cli && echo MENDER_CLI_REV_GIT_SHA=$(git rev-parse HEAD) >> ${CI_PROJECT_DIR}/build_revisions.env)
+    - git clone https://github.com/mendersoftware/mender-image-tests || true
+    - (cd mender-image-tests && git fetch -u -f origin ${MENDER_IMAGE_TESTS_REV}:pr && git checkout pr)
+    - (cd mender-image-tests && echo MENDER_IMAGE_TESTS_REV_GIT_SHA=$(git rev-parse HEAD) >> ${CI_PROJECT_DIR}/build_revisions.env)
+    - cat ${CI_PROJECT_DIR}/build_revisions.env
+    - tar -czf /tmp/workspace.tar.gz .
+    - mv /tmp/workspace.tar.gz ${CI_PROJECT_DIR}/workspace.tar.gz
+  artifacts:
+    expire_in: 24h
```

Comment:
Would it be good to have a bit more here? Unless it is extremely space consuming I would suggest maybe 2w or something. It's not uncommon for investigations to take a bit of time if you're interrupted by other things.

Same goes for other occurrences.

Reply:
Yeah, I see your point.

It is 350MB and I am not sure how the artifact storage is billed from GitLab. @mzedel any input here?

Reply:
If I understand the docs right, the limit per artifact and stage might be 2000mb and on Gitlab.com they can potentially live forever, while the default is 30days...
I don't mind increasing the expiry time until we are contacted about it, the only concern would be a "fair use" expiry date to be a good practice for an open source project.
Good point to support debugging with a bit more time and 2w should probably be good. If you agree, I'll adjust this across all my PRs.

Reply:
Good, thanks for checking it. Let's make 2w the default for all our artifacts. I'll fix here